### PR TITLE
dtspo:25 Automate Neuvector

### PR DESCRIPTION
**Pull request by mokainos**
automate Neuvector
https://tools.hmcts.net/jira/browse/DTSPO-25

### CODE REVIEW

File: k8s/namespaces/neuvector/neuvector.yaml [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+      - '{"config":{"name":"Idam","criteria":[{"key": "address", "value": "idam-api.platform.hmcts.net", "op": "="}]}}'
+    neuvector:
+      tag: 4.0.1
+      resources:
+        requests:
+          cpu: "100m"
+          memory: "2Gi"
+        limits:
+          cpu: 1
+          memory: 5Gi
```
Comment (timja) [inline k8s/namespaces/neuvector/neuvector.yaml:78]: https://github.com/hmcts/cnp-flux-config/blob/master/k8s/namespaces/neuvector/patches/prod/neuvector.yaml#L10-L11
```suggestion
          memory: 5Gi
          containerd:
            enabled: true
```

File: k8s/namespaces/neuvector/neuvector.yaml [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+    config:
+      autoScan: true
+    rules:
+      admission:
+        enable: true
+        mode: "protect"  # one of: monitor, protect
+        rules:
+          - '{"config":{"id":1000,"category":"Kubernetes","comment":"Allow system namespaces","criteria":[{"name":"namespace","op":"containsAny","value":"neuvector,kube-system,knode-system"}],"disable":false,"rule_type": "exception","cfg_type":"user_created"}}'
+          - '{"config":{"id":1001,"category":"Kubernetes","comment":"Allow admin namespaces","criteria":[{"name":"namespace","op":"containsAny","value":"admin,kured,monitoring"}],"disable":false,"rule_type": "exception","cfg_type":"user_created"}}'
+          - '{"config":{"id":1002,"category":"Kubernetes","comment":"Allow HMCTS registries","criteria":[{"name":"imageRegistry","op":"containsAny","value":"https://hmctspublic.azurecr.io/,https://hmctsprivate.azurecr.io/"}],"disable":false,"rule_type": "exception","cfg_type":"user_created"}}'
```
Comment (timja) [inline k8s/namespaces/neuvector/neuvector.yaml:29]: also needs to allow the sds registries

File: k8s/namespaces/neuvector/namespace.yaml [hunk 3]
```diff
@@ -0,0 +1,5 @@
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: neuvector
+---
```
Comment (timja) [inline k8s/namespaces/neuvector/namespace.yaml:5]:
```suggestion
```

File: k8s/environments/sbox/common-overlay/neuvector/kustomization.yaml [hunk 4]
```diff
@@ -0,0 +1,12 @@
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+resources:
+- ../../../../namespaces/neuvector
+
+patches:
+- target:
```
Comment (timja) [inline k8s/environments/sbox/common-overlay/neuvector/kustomization.yaml:7]: shouldn't this just be a `patchesStrategicMerge`? https://github.com/hmcts/cnp-flux-config/blob/master/k8s/prod/cluster-00-overlay/kustomization.yaml#L44

File: k8s/environments/sbox/common-overlay/neuvector/kustomization.yaml [hunk 5]
```diff
@@ -0,0 +1,12 @@
+apiVersion: kustomize.config.k8s.io/v1beta1
```
Comment (timja) [inline k8s/environments/sbox/common-overlay/neuvector/kustomization.yaml:1]: why is this a neuvector kustomization and not an environment one?

and also we prefer to keep the kustomization's per cluster so that we can upgrade one cluster a t a time

File: k8s/namespaces/neuvector/fluentbit.yaml [hunk 6]
```diff
@@ -0,0 +1,32 @@
+---
```
Comment (timja) [inline k8s/namespaces/neuvector/fluentbit.yaml:1]: missing the secret? https://github.com/hmcts/cnp-flux-config/blob/master/k8s/sandbox/common/neuvector/fluentbit-log.yaml

File: k8s/environments/sbox/common-overlay/neuvector/kustomization.yaml (hunk 4 above)
Comment (mokainos) [reply k8s/environments/sbox/common-overlay/neuvector/kustomization.yaml]: modified accordingly

File: k8s/namespaces/neuvector/patches/stg/cluster-00/neuvector.yaml [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+                - security-event
+                - audit
+              Syslog_In_Json: false
+              Cluster_Name: stg00
+      manager:
+        ingress:
+          host: neuvector00.staging.platform.hmcts.net
+          enabled: true
+          annotations:
+            kubernetes.io/ingress.class: traefik-neuvector-api
```
Comment (timja) [inline k8s/namespaces/neuvector/patches/stg/cluster-00/neuvector.yaml:43]:
```suggestion
            kubernetes.io/ingress.class: traefik-neuvector-api

```

File: k8s/namespaces/neuvector/patches/stg/cluster-01/neuvector.yaml [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+                - security-event
+                - audit
+              Syslog_In_Json: false
+              Cluster_Name: stg01
+      manager:
+        ingress:
+          host: neuvector01.staging.platform.hmcts.net
+          enabled: true
+          annotations:
+            kubernetes.io/ingress.class: traefik-neuvector-api
```
Comment (timja) [inline k8s/namespaces/neuvector/patches/stg/cluster-01/neuvector.yaml:43]:
```suggestion
            kubernetes.io/ingress.class: traefik-neuvector-api

```

File: k8s/namespaces/neuvector/patches/stg/cluster-00/neuvector.yaml (hunk 7 above)
Comment (timja) [inline k8s/namespaces/neuvector/patches/stg/cluster-00/neuvector.yaml:43]: I think this requires another traefik instance deployed? you probably want to do that first in a separate PR? or we can merge this but you won't be able to connect to it via ingress?

**Review by timja — APPROVED** [review]
